### ISSUE TITLE
Provide Fedora based devcontainer

### ISSUE THREAD

**Issue by mwisnicki** [microsoft/vscode-dev-containers#1292]
Please provide alternative devcontainer based on Fedora. Sometimes developing on Ubuntu is not an option.

Relates to: <Codespaces | Remote - Containers | Both>

**Comment by eitsupi** [comment]
I think you may build it yourself by using `common-redhat.sh`.
https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/common.md

**Comment by mwisnicki** [comment]
Thanks for pointing me to that script. I did it by hand without knowing about it. Would be nice if this was more documented and official image would be even better.

**Comment by Chuxel** [comment]
Given the nuances involved, we've optimized for the debian/ubuntu distro tree for our pre-built images and samples. We do not currently have plans to expand outside of this, but it would be interesting to hear about the scenarios you are thinking about where developing on Debian/Ubuntu is not an option.

To be clear, the extension itself is not limited to only these distributions at all, however, and we are looking into how we can enable the community more easily share definitions and features like those in this repository (as mentioned in https://github.com/microsoft/vscode-dev-containers/issues/1291). We want to make it easier for everyone to collaborate than is possible with this repository alone.

**Comment by mwisnicki** [comment]
And this is exactly the use case I consider to be the strong point of CodeSpaces - it doesn't matter which distro or even platform I personally use (which would be Windows 11🙂) because I can quickly get any environment tailored to specific needs of the project at hand.

**Comment by Chuxel** [comment]
@mwisnicki Yep agreed. You can encapsulate those setup instructions in a Dockerfile like you are saying. Given those instructions existed, was it the common script that you were looking for?

**Comment by mwisnicki** [comment]
I looked for any kind of instructions. It wasn't even clear from documentation whether you can use any non-codespaces aware base image (turns out you can just base it on fedora:latest and things just work even without common script).

I used google(github codespaces fedora) and google(github codespaces base image) and didn't find anything on the first pages of results. Now it's finding this ticket so should be easier :)

I think it would be useful to update https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/configuring-codespaces-for-your-project#dockerfile to mention you can use any base image, not necessarily from mcr.microsoft.com/vscode/devcontainers/*.

I'm sure one could follow all the links in documentation and find this information somewhere after reading dozens of pages of text but there's just way too much of it.

Of course ideal experience is not having to reach out to documentation/google at all and just have it offered by codespaces UI before creating environment, like you offer Alpine.

**Comment by Chuxel** [comment]
@bamurtaugh This is good feedback re docs 👆

We definitely don't want to leave the impression that you have to use our images - they're there as a convenience for when you don't have one already you want to build up from.

That said, for a base image, makes sense. Flipping to a feature request for that bit.

**Comment by bamurtaugh** [comment]
This is great feedback, thanks @mwisnicki!

Look at [docs on the VS Code side](https://code.visualstudio.com/docs/remote/containers#_create-a-devcontainerjson-file), it looks like we mention `You can use any image, Dockerfile, or set of Docker Compose files as a starting point. Here is a simple example that uses one of the pre-built [VS Code Development Container images](https://hub.docker.com/_/microsoft-vscode-devcontainers):`, which seems to help with this feedback (please let me know if it could be clearer!).

@2percentsilk maybe in the Codespaces doc, we could add something similar? i.e. amend the text to:

> You can add a Dockerfile to your project to define a container image and install software. In the Dockerfile, you can use FROM to specify the container image**, which may be hosted in any registry**.

**Comment by mwisnicki** [comment]
Yes that's pretty useful but does not come up when google search terms include "github" :(
I'm pretty spoiled by usually finding answer to my questions within first few results and often highlighted by google :)
